### DISCUSSION TITLE
b'CPU : Item not available' Error

### DISCUSSION THREAD

**Discussion by KAYAUgurBerat** [gijzelaerr/python-snap7#323] [Q&A] [answered]
Hello everyone, I want to change the values of the data blocks in the plc with the snap7 library. When I try this, I get the error shown below. 

b'CPU : Item not available'
Traceback (most recent call last):
  File "D:/Belgeler D/pythonProject/plcdeneme.py", line 10, in <module>
    plc_info = plc.get_cpu_info()
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\site-packages\snap7\client.py", line 161, in get_cpu_info
    check_error(result, context="client")
  File "C:\Program Files (x86)\Microsoft Visual Studio\Shared\Python37_64\lib\site-packages\snap7\common.py", line 92, in check_error
    raise Snap7Exception(error)
snap7.exceptions.Snap7Exception: b'CPU : Item not available'

Process finished with exit code 1

And this is my code:

```
import snap7
IP = "192.168.0.1"
RACK = 0
SLOT = 1

plc = snap7.client.Client()
plc.connect(IP,RACK,SLOT)

plc_info = plc.get_cpu_info()
print(f"adı: {plc_info.ModuleTypeName}")

state = plc.get_cpu_state()
print(f"State: {state}")
```

I made all the plc settings on Snap7's own site. What should I do to fix this problem? Thank you from n
![plc2](https://user-images.githubusercontent.com/81776415/137928460-7c5409a4-00b0-42a3-887d-3efbd9b55440.PNG)
![plc1](https://user-images.githubusercontent.com/81776415/137928465-bd824c41-ce0a-4703-a7be-fdc863b730aa.PNG)
ow

**Comment by swamper123** [comment]
This error can be legit...
What PLC do you got?

**Reply by KAYAUgurBerat** [reply]
I'm using S7-1200

**Reply by swamper123** [reply]
Sorry for the very late response. :sweat_smile: 
 
What I know is, that the 1200 and 1500 are not supporting all features.
I am sorry, but atm I haven't got a device on hand to verify this.

Since this doesn't seem to be an Python issue, you may ask at the [snap7 directly.](https://sourceforge.net/p/snap7/discussion/general/). Davide is very helpfull and just use his great work as a python-wrapper. :)

**Reply by ussefdesouky** [reply]
Its working on s7 1500 using PLCSIM Advance follow this link http://snap7.sourceforge.net/snap7_client.html#1200_1500

**Reply by swamper123** [reply]
The only suggestions I have left are:

- Firmware version on the machine (got a problem with wchars e.g.)
- still a snap7 thing (we receive a plausible message from snap7, perhaps problems are below? Right version installed?)
- the db is not global

Otherwise I can't help you atm. :confused:

**Comment by rhyddid72** [comment]
me too, but i don't know how to slove that. If you go through, can you give me some tips. Thank you too much

**Reply by nikteliy** [reply]
Hi @rhyddid72 
1. What PLC do you have?
2. Which function is causing the error?
if you provide a network trace(captured with tcpdump or Wireshark), it will be easier to find the cause

**Comment by gijzelaerr** [comment, marked as answer]
Closing as this is a known S7-1200/1500 limitation. Not all SZL functions are supported by these PLC models. Basic read/write operations (db_read, db_write) should work fine.
